app/models/user.rb

```diff
@@ -17,8 +17,34 @@ def authorized_to?(access_request)
     end
   end
 
+  def inventory_groups
+    inventory_groups_access = rbac_permissions.select do |permission|
+      next if permission.permission != Rbac::INVENTORY_GROUPS_READ
+      return Rbac::INVENTORY_GROUPS_GLOBAL if permission.resource_definitions == Rbac::INVENTORY_GROUPS_GLOBAL
+
+      true if supported_operation?(permission.resource_definitions.first[:attributeFilter])
+    end
+
+    inventory_groups_access.map do |permission|
+      operation_value(permission.resource_definitions.first[:attributeFilter])
+    end.flatten
+  end
+
   private
 
+  def supported_operation?(attribute_filter)
+    attribute_filter[:operation] == 'in' && attribute_filter[:key] == 'groups.id' ||
+      attribute_filter[:operation] == 'equal' && attribute_filter[:key] == 'group.id'
+  end
+
+  def operation_value(attribute_filter)
+    if attribute_filter[:operation] == 'in' && attribute_filter[:key] == 'groups.id'
+      attribute_filter[:value]
+    elsif attribute_filter[:operation] == 'equal' && attribute_filter[:key] == 'group.id'
+      [attribute_filter[:value]]
+    end
+  end
```

Comment:
You're basically doing the same condition twice, once in the `select` once in the `map`. I think it makes sense to merge them somehow and do the condition just once.